src/plugins/iconv/iconv.c

```diff
@@ -184,8 +232,15 @@ int elektraIconvGet(Plugin *handle, KeySet *returned, Key *parentKey)
 		ksDel (pluginConfig);
 		return 1;
 	}
-
-	if (!kdbbNeedsUTF8Conversion(handle)) return 0;
```

Comment:
the problem is that if you mount iconv with from=UTF-8 and no to= or to=UTF-8 no validation is happening, thats why i removed it.